### PR TITLE
[NEW] Redesign authentication views

### LINKED ISSUE #1454: [NEW] Redesign Onboarding
screen shot 2018-07-09 at 11 15 45
screen shot 2018-07-09 at 11 15 52

### PR DESCRIPTION
**Pull request by philipbrito**
Closes https://github.com/RocketChat/Rocket.Chat.Android/pull/1504 
Closes #1454

### CODE REVIEW

**Review by rafaelks — CHANGES_REQUESTED** [review]
My review so far @filipedelimabrito:

1. WordPress icon is wrong.

screenshot_20180927-175914

2. Profile screen is empty.

screenshot_20180927-180618

3. Can't scroll to see the connect button while typing.

screenshot_20180927-175845

4. Google Smart Lock is super intrusive, doesn't allow me to hide it or go back.

screenshot_20180927-175938

**Comment by philipbrito** [comment]
@rafaelks Please note that this is a **WIP** PR.
For your point number four: Are we back again with the discussion about GSL? Btw, note that you can tap "NONE OF ABOVE" button to dismiss the dialog.

Please, let's review the UI (not the code) after the removing of the WIP on the title of the PR.
Thanks

**Comment by rafaelks** [comment]
@filipedelimabrito Please check our `android-design` channel in order to understand how GSL should work. Also, "NONE OF ABOVE BUTTON" doesn't work. I can record a video for you if you want.

**Comment by philipbrito** [comment]
@rafaelks Note that  you **CAN NOT** change the look and feel of the GSL! You're probably seeing that view because you have more than one account saved with GSL.

This is what I get for your reference:

gsl

Again, since it is a **WIP PR** let's discuss anything **about the UI** internally. 
Thanks.

File: app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+    private fun doAuthentication(loginType: Int) {
+        launchUI(strategy) {
+            view.showLoading()
+            try {
+                val token = retryIO("login") {
+                    when (loginType) {
+                        TYPE_LOGIN_OAUTH -> client.loginWithOauth(credentialToken, credentialSecret)
+                        TYPE_LOGIN_CAS -> {
+                            delay(3, TimeUnit.SECONDS)
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:109]: Why we need this delay?

File: app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                            client.loginWithCas(credentialToken)
+                        }
+                        TYPE_LOGIN_SAML -> {
+                            delay(3, TimeUnit.SECONDS)
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:113]: Also here?

File: app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt [hunk 3]
```diff
@@ -6,25 +6,39 @@ import android.os.Bundle
...
+import io.reactivex.disposables.Disposable
 import kotlinx.android.synthetic.main.fragment_authentication_register_username.*
+import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
-internal const val TAG_REGISTER_USERNAME_FRAGMENT = "RegisterUsernameFragment"
+private const val BUNDLE_USER_ID = "user_id"
+private const val BUNDLE_AUTH_TOKEN = "auth_token"
+
+fun newInstance(userId: String, authToken: String): Fragment {
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt:34]: Wouldn't be better if we put this method declaration in a `companion object` inside `RegisterUsernameFragment`?

File: app/src/main/java/chat/rocket/android/authentication/resetpassword/ui/ResetPasswordFragment.kt [hunk 4]
```diff
@@ -108,42 +116,35 @@ class ResetPasswordFragment : Fragment(), ResetPasswordView {
...
     private fun enableUserInput() {
-        button_reset_password.isEnabled = true
+        enableButtonConnect()
         text_email.isEnabled = true
     }
 
     private fun disableUserInput() {
-        button_reset_password.isEnabled = false
+        disableButtonConnect()
         text_email.isEnabled = true
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/authentication/resetpassword/ui/ResetPasswordFragment.kt]: Is this always `true`?

File: app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt [hunk 5]
```diff
@@ -188,47 +216,49 @@ class ServerFragment : Fragment(), ServerView {
...
+            if (url.scheme() == "https") {
+                spinner_server_protocol.setSelection(0)
+            } else {
+                spinner_server_protocol.setSelection(1)
+            }
 
-            val serverUrl = url.toString().removePrefix("${url.scheme()}://")
-            text_server_url.textContent = serverUrl
+            protocol = "${url.scheme()}://"
+            text_server_url.textContent = url.toString().removePrefix("${url.scheme()}://")
```
Comment (leonardoaramaki) [inline app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt]: I think you meant to put `text_server_url.textContent = url.toString().removePrefix(protocol)` :thinking:

File: app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt (hunk 1 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:109]: We need to add a delay to make the CAS authentication work properly. If we remove that we'll see a message like "Unable to authenticate". This delay is needed due to the CAS authentication processing.

File: app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt (hunk 2 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/authentication/loginoptions/presentation/LoginOptionsPresenter.kt:113]: We need to add a delay to make the SAML authentication work properly. If we remove that we'll see a message like "Unable to authenticate". This delay is needed due to the CAS authentication processing.

File: app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt (hunk 3 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/authentication/registerusername/ui/RegisterUsernameFragment.kt:34]: I'm not sure. `Companion object` works fine too, but doing that we'll need to make the same for all the others fragments.

File: app/src/main/java/chat/rocket/android/authentication/resetpassword/ui/ResetPasswordFragment.kt (hunk 4 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/authentication/resetpassword/ui/ResetPasswordFragment.kt]: 👀 👁  Thanks for the catch!
👍

File: app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt (hunk 5 above)
Comment (philipbrito) [reply app/src/main/java/chat/rocket/android/authentication/server/ui/ServerFragment.kt]: Good catch!!
👍

**Review by leonardoaramaki — APPROVED** [review]

**Review by rafaelks — APPROVED** [review]

**Review by luciofm — APPROVED** [review]

**Pull request merged**